src/Canvas.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import { Point } from './Point'
+import { Viewport } from './Viewport';
+import { emitKeypressEvents } from 'readline';
+
+interface ICanvasProps {
+  viewport: Viewport
+  updateViewport: (viewport: Viewport) => any,
+}
+
+interface ICanvasState {
+
+}
+
+export class Canvas extends React.Component <ICanvasProps, ICanvasState> {
+  svgRef: React.RefObject<SVGSVGElement>
+  gRef: React.RefObject<SVGGElement>
```

Comment:
not sure if we need `gRef`

Reply:
Not in the current code. But I would suggest we actually attach `onwheel` directly to the DOM instead of the react event so that we can cancel the event and avoid the browser scrolling when you zoom in. We'd need to be careful to manually tear that event handler down in a `componentWillUnmount` thing.